src/plugins/ai_assistant_management/observability/public/routes/components/knowledge_base_bulk_import_flyout.tsx

```diff
@@ -0,0 +1,168 @@
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCode,
+  EuiCodeBlock,
+  EuiFilePicker,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiHorizontalRule,
+  EuiIcon,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { KnowledgeBaseEntry } from '../../../common/types';
+import { useImportKnowledgeBaseEntries } from '../../hooks/use_import_knowledge_base_entries';
+
+export function KnowledgeBaseBulkImportFlyout({ onClose }: { onClose: () => void }) {
+  const { mutateAsync, isLoading } = useImportKnowledgeBaseEntries();
+
+  const filePickerId = useGeneratedHtmlId({ prefix: 'filePicker' });
+
+  const [files, setFiles] = useState<File[]>([]);
+
+  const onChange = (file: FileList | null) => {
+    setFiles(file && file.length > 0 ? Array.from(file) : []);
+  };
+
+  const handleSubmitNewEntryClick = async () => {
+    let entries: Array<Omit<KnowledgeBaseEntry, '@timestamp'>> = [];
+    const text = await files[0].text();
+
+    const elements = text.split('\n').filter(Boolean);
+
+    try {
+      entries = elements.map((el) => JSON.parse(el)) as Array<
+        Omit<KnowledgeBaseEntry, '@timestamp'>
+      >;
+    } catch (_) {
+      /* empty */
+    }
+
+    try {
+      await mutateAsync({ entries });
+      onClose();
+    } catch (_) {
+      /* empty */
+    }
+  };
+
+  return (
+    <EuiFlyout onClose={onClose}>
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle>
+          <h2>
+            {i18n.translate(
+              'aiAssistantManagementObservability.knowledgeBaseBulkImportFlyout.h2.bulkImportLabel',
+              { defaultMessage: 'Import files' }
+            )}
+          </h2>
+        </EuiTitle>
+      </EuiFlyoutHeader>
+
+      <EuiFlyoutBody>
+        <EuiFlexGroup alignItems="center">
+          <EuiFlexItem grow={false}>
+            <EuiIcon type="addDataApp" size="xl" />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiTitle size="xs">
+              <h3>
+                {i18n.translate(
+                  'aiAssistantManagementObservability.knowledgeBaseBulkImportFlyout.addFilesToEnrichTitleLabel',
+                  { defaultMessage: 'Add files to enrich your Knowledge base' }
+                )}
+              </h3>
+            </EuiTitle>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+
+        <EuiSpacer size="l" />
+
+        <EuiText size="s">
+          <FormattedMessage
+            id="aiAssistantManagementObservability.knowledgeBaseBulkImportFlyout.uploadAJSONFileTextLabel"
+            defaultMessage="Upload a newline delimited JSON ({ext}) file containing a list of entries to add to your Knowledge base."
+            values={{
+              ext: <EuiCode language="html">.ldjson</EuiCode>,
```

Comment:
```suggestion
              ext: <EuiCode language="html">.ndjson</EuiCode>,
```

Reply:
The suggestion was supposed to be committed. If that hasn't happened then that was by accident.